rclcpp_components/cmake/rclcpp_components_register_node.cmake

```diff
@@ -70,10 +70,11 @@ macro(rclcpp_components_register_node target)
   file(GENERATE OUTPUT ${PROJECT_BINARY_DIR}/rclcpp_components/node_main_${node}.cpp
     INPUT ${PROJECT_BINARY_DIR}/rclcpp_components/node_main_configured_${node}.cpp.in)
   add_executable(${node} ${PROJECT_BINARY_DIR}/rclcpp_components/node_main_${node}.cpp)
-  ament_target_dependencies(${node}
-    "rclcpp"
-    "class_loader"
-    "rclcpp_components")
+  target_link_libraries(${node}
+    class_loader::class_loader
+    rclcpp::rclcpp
+    rclcpp_components::component
```

Comment:
I think that `component` is recursively dependent on these other ones.  I imagine that you aren't looking to audit all of the dependencies in this PR, but maybe something that we should check/address in the future.

Reply:
While that's true, it is also the case that the node_main.cpp also directly depends on all of these dependencies:

https://github.com/ros2/rclcpp/blob/d08d9cf5ffd5ecb80c1ef4085573c411eecbd737/rclcpp_components/src/node_main.cpp.in#L19-L22

So I think having all of these dependencies here is appropriate.